packages/@react-stately/select/src/useSelectState.ts

```diff
@@ -15,7 +15,7 @@ import {SelectProps} from '@react-types/select';
 import {SingleSelectListState, useSingleSelectListState} from '@react-stately/list';
 import {useState} from 'react';
 
-export interface SelectState<T> extends SingleSelectListState<T>, MenuTriggerState {
+export interface SelectState<T> extends Omit<SingleSelectListState<T>, 'disallowEmptySelection'>, MenuTriggerState {
```

Comment:
Actually we should be omitting `disallowEmptySelection` from the `SelectProps` rather than the `SelectState` (this is actually the return type). We'll need to perform this `Omit` on the `SingleSelection` portion of https://github.com/adobe/react-spectrum/blob/efa050d23922f9a2b45848f4c2a644ec2e2caba0/packages/%40react-types/select/src/index.d.ts#L32 

If you'd like, you may want to remove `disallowEmptySelection` from https://github.com/adobe/react-spectrum/blob/main/packages/@react-stately/list/src/useSingleSelectListState.ts#L19 as well since that hook doesn't actually allow the user to modify that setting since we always provide `disallowEmptySelection: true` to `useListState`. There may be a couple of other hooks that consume `useSingleSelectListState` that ought to have the same omit as well

Reply:
Thank you for the feedback, I've updated the changes.